cmd/openshift-tests/storagecaps.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+)
+
+// Manifest yaml structs
+
+type StorageClass struct {
+	FromFile string `yaml:"FromFile"`
+}
+
+type SnapshotClass struct {
+	FromName bool `yaml:"FromName"`
+}
+
+type Capabilities struct {
+	Block    bool `yaml:"block"`
+	RWX      bool `yaml:"RWX"`
+	Snapshot bool `yaml:"snapshotDataSource"`
+}
+
+type DriverInfo struct {
+	Name         string `yaml:"Name"`
+	Capabilities `yaml:"Capabilities"`
+}
+
+type YamlManifest struct {
+	ShortName     string `yaml:"ShortName"`
+	StorageClass  `yaml:"StorageClass"`
+	SnapshotClass `yaml:"SnapshotClass"`
+	DriverInfo    `yaml:"DriverInfo"`
+}
+
+const (
+	testReportEnvVar = "TEST_REPORT"
+)
+
+var (
+	manifestFilename string
+)
+
+// Generic test reporting with env var selector
+func printReport() {
+	testReport := os.Getenv(testReportEnvVar)
+
+	if testReport == "storage-capabilities" && manifestFilename != "" {
+		printStorageCapabilities()
+	}
+}
+
+func printStorageCapabilities() {
+	yamlFile, err := ioutil.ReadFile(manifestFilename)
+	if err != nil {
+		fmt.Println("Failed to", err)
+		return
+	}
+
+	var yamlManifest YamlManifest
+	err = yaml.Unmarshal(yamlFile, &yamlManifest)
+	if err != nil {
+		fmt.Println("Error parsing", err)
+		return
+	}
+
+	supportsSnapshot := yamlManifest.DriverInfo.Capabilities.Snapshot && yamlManifest.SnapshotClass.FromName
+
+	fmt.Println()
```

Comment:
Commands are expected to take the stream to print to as an argument, pass an io.Writer in to this method and use `fmt.Fprintf`